### ISSUE TITLE
Unavailable iOS SDK version fails silently

### ISSUE THREAD

**Issue by sgleadow** [moredip/Frank#188]
I just noticed this on a fresh machine that didn't have iOS 6.0 SDK simulator version installed and trying to run Frank tests against 6.0. It just failed to launch the simulator, but didn't give any useful error. Took me a while to track down.

I'm not sure where would be the best place to add a check that the SDK we're trying to launch exists in the `xcodebuild` list of available SDKs, but it should be a pretty easy fix.

@moredip - any idea where the best place to add the check would be?

This is our current workaround in our build script, but would be better in Frank.

raise "iOS SDK #{@sdk} is not installed." unless `xcodebuild -showsdks` =~ /iOS #{@sdk}/

**Comment by conradwt** [comment]
I'm guessing that this information is best left for the general documentation as part of the installation requirements.

**Comment by sgleadow** [comment]
I'm not so sure... I run 3 different Macs for iOS development. Some of them have a lot of SDKs installed, others, not much at all. Our CI box particularly, we want to have lots of versions and run tests on all those versions.

The way it currently just stalls if you tell it to run the tests in a simulator SDK that isn't available seems misleading, when there is an easy one line fix to report it as an error. It's not a problem for me anymore, because it's in my Rakefile, but I can imagine hitting this issue on my next project.

Happy to bow to our benevolent dictator on this - @moredip thoughts?

**Comment by moredip** [comment]
I agree with Stew - if you explicitly ask to use an SDK which isn't available then Frank should fail fast with a descriptive error.

Cheers,

## Pete

Typed on a little bitty keyboard
